File: app/src/main/java/com/jnjcomu/edison/activity/LoginActivity.java
package com.jnjcomu.edison.activity;

import android.app.ProgressDialog;
import android.content.Intent;
import android.support.v7.app.AlertDialog;
import android.support.v7.app.AppCompatActivity;
import android.widget.EditText;

import com.jnjcomu.edison.R;
import com.jnjcomu.edison.api.APIBuilder;
import com.jnjcomu.edison.model.Ticket;

import org.androidannotations.annotations.AfterViews;
import org.androidannotations.annotations.Click;
import org.androidannotations.annotations.EActivity;
import org.androidannotations.annotations.UiThread;
import org.androidannotations.annotations.ViewById;

import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;

@EActivity(R.layout.activity_login)
public class LoginActivity extends AppCompatActivity {

    @ViewById(R.id.edt_id_field)
    protected EditText edtIdField;

    @ViewById(R.id.edt_pw_field)
    protected EditText edtPwField;

    protected ProgressDialog statusDialog;

    @AfterViews
    protected void initUI() {
        statusDialog = new ProgressDialog(this);
        statusDialog.setMessage("잠시만 기다려 주세요...");
    }

    @Click(R.id.btn_login)
    protected void doTryLogin() {
        String userId = edtIdField.getText().toString();
        String userPassword = edtPwField.getText().toString();

        startActivity(new Intent(this, MainActivity_.class));
        finish();
    }

    @UiThread
    protected void sendResult(boolean finishLogin, String message) {
        statusDialog.dismiss();

        if (finishLogin) {
            startActivity(new Intent(this, MainActivity_.class));
            finish();
        }

        if (!"".equals(message)) {
            new AlertDialog.Builder(this)
                    .setMessage(message)
                    .show();
        }
    }

    protected void sendResult(boolean finishLogin) {
        sendResult(finishLogin, "");
    }
}

File: app/src/main/java/com/jnjcomu/edison/activity/LoginActivity.kt
package com.jnjcomu.edison.activity

import android.content.Intent
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.view.View
import android.widget.Toast
import com.jnjcomu.edison.R
import com.jnjcomu.edison.addition.appStorage
import com.jnjcomu.edison.api.API
import com.jnjcomu.edison.broadcast.BootBroadcastReceiver
import com.jnjcomu.edison.model.Session
import kotlinx.android.synthetic.main.activity_login.*
import retrofit2.Call
import retrofit2.Callback
import retrofit2.Response

class  LoginActivity : AppCompatActivity(), View.OnClickListener {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_login)
    }

    override fun onClick(v: View) {
        progress_bar.visibility = View.VISIBLE
        progress_bar.isIndeterminate = true

        val call = API.getApi(this).login(edt_id_field.text.toString(), edt_pw_field.text.toString())

        call.enqueue(object : Callback<Session> {
            override fun onResponse(call: Call<Session>?, response: Response<Session>?) {
                progress_bar.visibility = View.INVISIBLE
                try {
                    if (response?.body()?.login.equals("true")) {
                        appStorage.saveUserName(response!!.body()!!.name)
                        if(appStorage.isFirstRun)
                            firstRun()
                        else {
                            startActivity(Intent(applicationContext, CheckInActivity::class.java))
                            finish()
                        }
                    } else {
                        showToast("잘못된 아이디나 비밀번호입니다.")
                    }
                } catch (e: NullPointerException) {
                    showToast("잘못된 아이디나 비밀번호입니다.")
                }
            }

            override fun onFailure(call: Call<Session>?, t: Throwable?) {
                progress_bar.visibility = View.INVISIBLE
                showToast("잘못된 아이디나 비밀번호입니다.")
            }
        })
    }

    fun firstRun() {
        sendBroadcast(Intent(this, BootBroadcastReceiver::class.java))
        startActivity(Intent(this, IntroActivity::class.java))
        finish()
    }

    fun showToast(msg: String) {
        Toast.makeText(applicationContext, msg, Toast.LENGTH_SHORT).show()
    }
}
